### PR TITLE
ICU-21945 Add support for Visual Studio 2022

### PR DESCRIPTION
**Pull request by chetanladdha**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21945
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by jefgen** [comment]
Thanks @chetanladdha!

Can we also maybe consider adding a CI build bot that uses VS2022 in the https://github.com/unicode-org/icu/blob/main/.ci-builds/.azure-pipelines.yml file as well? That would give us build/test coverage as well.

Or perhaps that could be another separate ticket?

**Comment by erik0686** [comment]
I'm wondering if we need to add a condition to line 38, to add v143 as well :) 

`<PropertyGroup Condition="'$(WindowsTargetPlatformVersion)'=='' and ( '$(PlatformToolset)'=='v142' or  '$(PlatformToolset)'=='v143')">
    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
  </PropertyGroup>`

**Comment by jefgen** [comment]
> I'm wondering if we need to add a condition to line 38, to add v143 as well :)

Good catch!

File: icu4c/source/allinone/Build.Windows.PlatformToolset.props [hunk 1]
```diff
@@ -13,10 +13,12 @@
       v140 is the Visual Studio 2015 toolset. (14.0)
       v141 is the Visual Studio 2017 toolset. (15.0)
       v142 is the Visual Studio 2019 toolset. (16.0)
+	  v143 is the Visual Studio 2022 toolset. (17.0)
```
Comment (jefgen) [inline icu4c/source/allinone/Build.Windows.PlatformToolset.props:16]: nit: mixing tabs and spaces on the same line.
(Please use spaces only here)

**Comment by erik0686** [comment]
@chetanladdha friendly FYI :) as a general practice for ICU Pull Requests, the Pull Request owner usually assigns someone (not themselves) to the PR. In this case, you can probably assign myself, or Jeff or Daniel, since we're familiar with the changes being made. Otherwise this PR will show up on the list of PRs without assignee on the TC minute document. You can see this option below the "reviewers" section on the right :)

**Review by erik0686 — APPROVED** [review]

**Comment by jefgen** [comment]
Note: You'll also need to sign the CLA as well before this can be merged.

**Comment by jefgen** [comment]
Regarding the CLA:
It looks like the commit was created using the "@microsoft.com" email address. I suspect what might be happening is that your GitHub profile doesn't have this email address associated with the account.
If you add this email address to your account, then I think it will work.

**Comment by jefgen** [comment]
Gentle bump to take a look at the CLA.

**Review by jefgen — APPROVED** [review]
lgtm, thanks Chetan!

**Pull request merged**